### PR TITLE
 📖 Automate the release links in mdbook

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
This PR automates the release notes links in the mdbook so that they get regenerated using the latest version in the range specified when building it.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -80,7 +78,7 @@ Check the [vSphere provider releases](https://github.com/kubernetes-sigs/cluster
 For more information about prerequisites, credentials management, or permissions for vSphere, visit the [getting started guide](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/docs/getting_started.md).
 
 ```bash
-kubectl create -f https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/releases/download/v0.5.0/infrastructure-components.yaml
+kubectl create -f {{#releaselink gomodule:"sigs.k8s.io/cluster-api-provider-vsphere" asset:"infrastructure-components.yaml" version:"0.5.x"}}
```
Comment (chuckha) [inline docs/book/src/user/quick-start.md]: now we just need a prepreprocessor to process the right version numbers before the preprocessor runs 😆

File: hack/tools/mdbook/releaselink/releaselink.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+			return "", err
+		}
+
+		rawURL := url.URL{
+			Scheme: "https",
+			Host:   "proxy.golang.org",
+			Path:   path.Join(gomodule, "@v", "/list"),
+		}
+
+		resp, err := http.Get(rawURL.String())
```
Comment (chuckha) [inline hack/tools/mdbook/releaselink/releaselink.go]: Where in the pipeline does release happen?
If it's inconvenient for the preprocessor to fail you might consider a retry here. But if it's not a big deal to simply re-run then this is not an issue

Comment (vincepri) [reply hack/tools/mdbook/releaselink/releaselink.go]: If it fail, it should be ok to just retry, it doesn't take long in CI (<1m) or locally (<2s)

**Comment by vincepri** [comment]
@chuckha I added the godoc to the struct, the choice of go struct tags is the fastest way to parse strings without adding more external dependencies, or use json. It's also used elsewhere and it's pretty straightforward.

**Pull request merged**